docs/design/federated-services.md

```diff
@@ -0,0 +1,490 @@
+# Kubernetes Cluster Federation (a.k.a. "Ubernetes")
+## Cross-cluster Load Balancing and Service Discovery 
+### Requirements and System Design
+### by Quinton Hoole, Dec 3 2015
+
+## Requirements
+
+### Discovery, Load-balancing and Failover
```

Comment:
No, not at all.  I would put it at number 4 on the priority list, below GCE, AWS and basic bare metal.  I don't know too much about Azure, so it was expedient to focus on the first three, about which I do know something :-)